Modules/CTP/src/RawDataQcTask.cxx

```diff
@@ -0,0 +1,224 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   RawDataQcTask.cxx
+/// \author Marek Bombara
+///
+
+#include <TCanvas.h>
+#include <TH1.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "CTP/RawDataQcTask.h"
+#include "DetectorsRaw/RDHUtils.h"
+#include "Headers/RAWDataHeader.h"
+#include "DPLUtils/DPLRawParser.h"
+#include "DataFormatsCTP/Digits.h"
+#include <Framework/InputRecord.h>
+#include <Framework/InputRecordWalker.h>
+
+namespace o2::quality_control_modules::ctp
+{
+
+RawDataQcTask::~RawDataQcTask()
+{
+  delete mHistoBC;
+  delete mHistoInputs;
+  delete mHistoClasses;
+}
+
+void RawDataQcTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info, Support) << "initialize RawDataQcTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
+  if (auto param = mCustomParameters.find("myOwnKey"); param != mCustomParameters.end()) {
+    ILOG(Info, Devel) << "Custom parameter - myOwnKey: " << param->second << ENDM;
+  }
+
+  mHistoBC = new TH1F("histobc", "BC distribution", 3564, 0, 3564);
+  mHistoInputs = new TH1F("inputs", "Inputs distribution", 48, 0, 48);
+  mHistoClasses = new TH1F("classes", "Classes distribution", 64, 0, 64);
+  getObjectsManager()->startPublishing(mHistoBC);
+  getObjectsManager()->startPublishing(mHistoInputs);
+  getObjectsManager()->startPublishing(mHistoClasses);
+  try {
+    getObjectsManager()->addMetadata(mHistoBC->GetName(), "custom", "34");
```

Comment:
this is not necessary

Reply:
I do not know what those lines do, I thought they might be important. Is it ok to delete whole try { } catch { } section (lines 54-65)?

Reply:
yes